AmplifyPlugins/Core/AWSPluginsCore/Keychain/KeychainStore.swift

```diff
@@ -148,14 +148,26 @@ public struct KeychainStore: KeychainStoreBehavior {
         let fetchStatus = SecItemCopyMatching(getQuery as CFDictionary, nil)
         switch fetchStatus {
         case errSecSuccess:
+            #if os(macOS)
+            // Never update an existing keychain item on OS X, since the existing item could have unauthorized apps in the Access Control List. Fixes zero-day Keychain vuln found here: https://drive.google.com/file/d/0BxxXk1d3yyuZOFlsdkNMSGswSGs/view
+            SecItemDelete(getQuery as CFDictionary)
+            var attributesToSet = attributes.defaultSetQuery()
```

Comment:
Since the remaining code is the same as the `errSecItemNotFound` case, could we just `fallthrough` instead?

Reply:
Updated per suggestion.